### ISSUE TITLE
Docker Hub setup

### ISSUE THREAD

**Issue by LBegnaud** [netdisco/netdisco-docker#2]
Hey @ollyg I'm getting close to a finished product in my fork.

I still have some ideas to implement, but I believe the image in my repo is workable:

https://hub.docker.com/r/lbegnaud/netdisco-docker/
https://github.com/LBegnaud/netdisco-docker

Probably a good time to start setting up the docker hub side of things for the netdisco org. I'm not sure how that works, but i can do some research if you'd like (I'm not sure if you can create an organization-type thing in docker hub, and I'm not sure if you can point docker hub directly to the github org or if it has to go through a personal account.). 

For a single account like mine the process is as such:

- create a docker hub account: https://hub.docker.com/
- go to account settings and link a github account
- Create an "Automated Build" with mostly defaults and point it to the netdisco-docker repo
- at this point docker hub is supposed to see new versions of the code and initiate a new build, but for my account I had to go into github and enable a hook in the other direction
  - Log into github
  - go to repo
  -  In settings, install the github service for docker
  ![image](https://user-images.githubusercontent.com/15981748/31180725-fb02d506-a8e4-11e7-9f0a-faf69d85b637.png)
 - At this point, everytime a branch is created or updated, docker hub initiates a new build of the image. 
  - it is also possible to setup build triggers on tags, but I'm not sure that's necessary for what we are doing here.

I need to clean up the documentation of the project obviously, but I figured I would give you some lead time to do the setup. I can push my code so far for you to test the workflow. Maybe put it into a test branch on the official repo? I'm not well-versed in git collaboration. So please let me know how you want to proceed.

**Comment by LBegnaud** [comment]
@ollyg 
FYI i've done some more work with this on other projects I'm working on internally. Looks like docker hub can be orchestrated by docker cloud. An org can be setup and tied to a github org (via ssh keys generated by a user's integration)

Long story short: we can create a docker org, tie it to this org, and setup automated builds. Should be pretty straightforward from https://cloud.docker.com , but let me know if you want me to do anything to that end.

**Comment by ollyg** [comment]
Okay! We are getting there...

https://cloud.docker.com/swarm/netdisco/repository/docker/netdisco/netdisco-docker/general
https://store.docker.com/community/images/netdisco/netdisco-docker

**Comment by ollyg** [comment]
I would like to set up a webhook from the main netdisco repo and have the docker image rebuilt when there's a new tag or release. It seems straightforward and other people are already doing this.

**Comment by ollyg** [comment]
Note to self: metadata labels are now available from MicroBadger:

https://microbadger.com/images/netdisco/netdisco-docker

**Comment by LBegnaud** [comment]
yep just tested the workflow and this works as expected:
1. Create triggers on docker repo under "configure automated builds"
  - ![image](https://user-images.githubusercontent.com/15981748/33837138-0daecf10-de51-11e7-86e5-5dfd63a77192.png)
2. create [webhook](https://help.github.com/articles/about-webhooks/) on the repo you'd like to trigger new docker builds (using the url from the first step)
  - 
![image](https://user-images.githubusercontent.com/15981748/33837176-28754c66-de51-11e7-8d59-fff1b2011b88.png)


I've created a pair of triggers in the `docker-netdisco` docker repo for you (step 1). I assume we want to at least tie those two repositories to the build process.

**Comment by ollyg** [comment]
Thanks! I changed things slightly - I have set up just one webhook from the netdisco/netdisco-docker repo to the docker cloud, and attached it to the 'create' action which is triggered when we add a tag.

My understanding of the Dockerfile is that it uses CPAN sources for Netdisco so attaching a webhook to the netdisco git repo would not be useful. Also it downloads netdisco-mibs Releases so again the webhook is not needed on that repo. We can instead simply `git tag` the netdisco/netdisco-docker repo when we push a new release (netdisco or snmp-info) that we want to have built into a new docker image.

In the future we could create a netdisco/netdisco-docker-bleed project in Docker cloud, to take the actual git HEAD code and build images of that.
